streams/dshr_strdata_mod.F90

```diff
@@ -1373,39 +1485,54 @@ subroutine shr_strdata_readLBUB(sdat, ns, mDate, mSec, newData, istr, rc)
 
     ! if model current date is outside of model lower or upper bound - find the stream bounds
     find_bounds = (rDateM < rDateLB .or. rDateM >= rDateUB)
-    if (debug > 0 .and. sdat%mainproc) then
-       write(sdat%stream(1)%logunit,'(a,i4,2x,6(i18,2x),l7)')' stream,lbymd,lbsec,mdate,msec,ubymd,ubsec,newdata= ',ns,&
-            sdat%pstrm(ns)%ymdLB,sdat%pstrm(ns)%todLB, &
-            mdate,msec, &
-            sdat%pstrm(ns)%ymdUB,sdat%pstrm(ns)%todUB,find_bounds
-       write(sdat%stream(1)%logunit,'(a,i4,2x,3(f20.3,2x),l7)')' stream,rdateLB,rdateM,rdateUB,newdata= ',&
-            ns,rdateLB,rdateM,rdateUB,find_bounds
-    end if
     if (find_bounds) then
        call ESMF_TraceRegionEnter(trim(istr)//'_fbound')
-       call shr_stream_findBounds(stream, mDate, mSec,  sdat%mainproc, &
+       call shr_stream_findBounds(stream, mDate, mSec, &
             sdat%pstrm(ns)%ymdLB, dDateLB, sdat%pstrm(ns)%todLB, n_lb, filename_lb, &
             sdat%pstrm(ns)%ymdUB, dDateUB, sdat%pstrm(ns)%todUB, n_ub, filename_ub)
        call ESMF_TraceRegionExit(trim(istr)//'_fbound')
-       if (debug > 0 .and. sdat%mainproc) then
-          write(sdat%stream(1)%logunit,'(a,i4,2x,6(i18,2x),l7)')' stream,lbymd,lbsec,mdate,msec,ubymd,ubsec,newdata= ',ns,&
-               sdat%pstrm(ns)%ymdLB,sdat%pstrm(ns)%todLB,&
-               mdate,msec, &
-               sdat%pstrm(ns)%ymdUB,sdat%pstrm(ns)%todUB
-          write(sdat%stream(1)%logunit,'(a,i4,2x,3(f20.3,2x),l7)')' stream,rdateLB,rdateM,rdateUB,newdata= ',&
-               ns,rdateLB,rdateM,rdateUB,find_bounds
-       end if
-    endif
+    end if
 
     ! determine if need to read in new stream data
     newdata = (sdat%pstrm(ns)%ymdLB /= oDateLB .or. sdat%pstrm(ns)%todLB /= oSecLB)
+
+    ! write time bounds info
+    if (debug_level > 0 .and. sdat%mainproc) then
+       if (debug_level > 0) then
```

Comment:
Is this statement made redundant by the statement above? I think you could remove this `if` -- `end if` pair.

Reply:
done